bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/ContentFragmentUtils.java

```diff
@@ -245,19 +252,32 @@ public static String getGridResourceType(Resource fragmentResource) {
      * @param resourceIterator Iterator of resources for which to get the component exporters.
      * @param modelFactory Model factory service.
      * @param slingHttpServletRequest Current request.
+     * @param callerResource The caller (XF) resource.
      * @return Ordered map of resource names to {@link ComponentExporter} models.
      */
     @NotNull
     public static LinkedHashMap<String, ComponentExporter> getComponentExporters(
         @NotNull final Iterator<Resource> resourceIterator,
         @NotNull final ModelFactory modelFactory,
-        @NotNull final SlingHttpServletRequest slingHttpServletRequest) {
+        @NotNull final SlingHttpServletRequest slingHttpServletRequest,
+        @NotNull final Resource callerResource) {
         final LinkedHashMap<String, ComponentExporter> componentExporterMap = new LinkedHashMap<>();
 
+        SlingHttpServletRequest wrappedSlingHttpServletRequest = new SlingHttpServletRequestWrapper(slingHttpServletRequest) {
+
+            @Override
+            public Object getAttribute(String name) {
+                if (ATTR_RESOURCE_CALLER_PATH.equals(name)) {
+                    String resourceCallerPath = (String)super.getAttribute(ATTR_RESOURCE_CALLER_PATH);
```

Comment:
Why is this line needed? In case this request attribute is set in a custom component?

Reply:
For HTML rendering, for XFs we only use the root-most resource path (obtained through the `componentContext`) concatenated with the path of the component inside the leaf-most XF (in case of nesting). I added a comment bellow explaining why we return the existing caller path in 4d14728.